t/test_script__program_runs.t

```diff
@@ -0,0 +1,248 @@
+use Test2::V0 -no_srand => 1;
+use Test::Script;
+use File::Temp qw( tempdir );
+use Data::Dumper qw( Dumper );
+use Probe::Perl;
+
+# Use the Perl interpreter as the program since it's the only one we know
+# exists.  The files in t/bin do not use any modules, so we don't have
+# to worry about passing @INC down.
+my $perl = Probe::Perl->find_perl_interpreter() or die "Can't find perl";
+
+subtest 'no interpreter_options' => sub {
+  like(
+    dies { program_runs $perl, { interpreter_options => 1 } },
+    qr{interpreter_options},
+    'program_runs rejects {interpreter_options => ...}'
+  );
+};
+
+subtest 'good' => sub {
+
+  subtest 'default name' => sub {
+
+    my $rv;
+    my $events;
+
+    is(
+      $events = intercept { $rv = program_runs [$perl, 't/bin/good.pl'] },
+      array {
+        event Ok => sub {
+          call pass => T();
+          call name => "Program $perl runs";
+        };
+        end;
+      },
+      'program_runs: perl t/bin/good.pl',
+    );
+
+    diag Dumper($events) unless $rv;
+
+    is $rv, T(), 'program_runs returns true as convenience';
+
+  };
+
+  subtest 'custom name' => sub {
+
+    my $rv;
+    my $events;
+
+    is(
+      $events = intercept { $rv = program_runs [$perl, 't/bin/good.pl'], 'It worked' },
+      array {
+        event Ok => sub {
+          call pass => T();
+          call name => 'It worked';
+        };
+        end;
+      },
+      'program_runs perl t/bin/good.pl It worked',
+    );
+
+    diag Dumper($events) unless $rv;
+
+    is $rv, T(), 'program_runs returns true as convenience';
+
+  };
+
+
+};
+
+subtest 'bad: returns 4' => sub {
+
+  subtest 'default name' => sub {
+
+    my $rv;
+    my $events;
+
+    my $rv2 = is(
+      $events = intercept { $rv = program_runs [$perl, 't/bin/four.pl'] },
+      array {
+        event Ok => sub {
+          call pass => F();
+          call name => "Program $perl runs";
+        };
+        event Diag => sub {};
+        event Diag => sub {
+          call message => match qr{4 - (?:Using.*\n# )?Standard Error\n};
+        };
+        end;
+      },
+      'program_runs perl t/bin/good.pl',
+    );
+
+    diag Dumper($events) unless $rv2;
+
+    is $rv, F(), 'program_runs returns false as convenience';
+
+  };
+
+  subtest 'custom name' => sub {
+
+    my $rv;
+    my $events;
+
+    my $rv2 = is(
+      $events = intercept { $rv = program_runs [$perl, 't/bin/four.pl'], 'It worked' },
+      array {
+        event Ok => sub {
+          call pass => F();
+          call name => 'It worked';
+        };
+        event Diag => sub {};
+        event Diag => sub {
+          call message => match qr{4 - (?:Using.*\n# )?Standard Error\n};
+        };
+        end;
+      },
+      'program_runs perl t/bin/good.pl It worked',
+    );
+
+    diag Dumper($events) unless $rv2;
+
+    is $rv, F(), 'program_runs returns false as convenience';
+
+  };
+
+
+};
+
+subtest 'unreasonable number of libs' => sub {
```

Comment:
we can skip this subtest for `program_runs` since it only relates to running a Perl script.